packages/permissionless/errors/gas.ts

```diff
@@ -0,0 +1,107 @@
+import { BaseError } from "viem"
+
+export class VerificationGasLimitNotEnough extends BaseError {
+    static message = /aa4[01]/
+    override name = "VerificationGasLimitNotEnough"
+    constructor({
+        cause,
+        verificationGasLimit,
+        docsPath
+    }: {
+        cause?: BaseError
+        verificationGasLimit?: bigint
+        docsPath?: string
+    }) {
+        super(
+            [
+                `Smart account and paymaster verification exceeded the verificationGasLimit ${verificationGasLimit} set for the user operation.`,
+                "",
+                "Possible solutions:",
+                "• Verify that the verificationGasLimit set for the user operation is high enough to cover the gas used during smart account and paymaster verification.",
+                "• If you are using the eth_estimateUserOperationGas or pm_sponsorUserOperation method from bundler provider like Pimlico to set user operation gas limits and the EntryPoint throws this error during submission, reach out to them.",
+                "",
+                docsPath ? `Docs: ${docsPath}` : ""
+            ].join("\n"),
+            {
+                cause
+            }
+        )
+    }
+}
+
+export class FundsLowerThanActualGasCost extends BaseError {
+    static message = /aa51/
+    override name = "FundsLowerThanActualGasCost"
+    constructor({
+        cause,
+        docsPath
+    }: {
+        cause?: BaseError
+        docsPath?: string
+    }) {
+        super(
+            [
+                "The actual gas cost of the user operation ended up being higher than the funds paid by the smart account or the paymaster.",
+                "",
+                "Possible solutions:",
+                "• If you encounter this error, try increasing the verificationGasLimit set for the user operation.",
+                "• If you are using the eth_estimateUserOperationGas or pm_sponsorUserOperation method from bundler provider like Pimlico to set user operation gas limits and the EntryPoint throws this error during submission, reach out to them.",
```

Comment:
I'd remove `like Pimlico`

Reply:
I'd actually remove it anywhere we mention it

Reply:
I thought over it, but yeah it makes sense to remove it